cpp/oneapi/dal/algo/kmeans/backend/gpu/kernels_fp.hpp

```diff
@@ -0,0 +1,461 @@
+/*******************************************************************************
```

Comment:
Where is problem with the current approach for kernels. You have:
- `kernels_fp.hpp`, where kernels declared and **defined** (implemented).
- `train_kernel_lloyd_dense_dpc.cpp` and `infer_kernel_dpc.cpp` that include `kernels_fp.hpp`:
- The functions defined in `kernels_fp.hpp` would be compiled every time once those files are compiled.

Correct approach:
- `kernels_fp.hpp` must contains only declarations.
- Introduce `kernels_fp_impl.hpp` that contains implementations of the functions declared in `kernels_fp.hpp` 
- Include `kernels_fp_impl.hpp` to `kernels_fp_impl_flt_dpc.cpp` and `kernels_fp_impl_dbl_dpc.cpp`
- Include  `kernels_fp.hpp` to other files, where functions are needed

Reply:
done